### PR TITLE
Do Topic Starter sort in PHP

### PR DESCRIPTION
**Pull request by albertlast**
I think it's a better approach to do the sort in php as in rdbms
- less db specific code
- less loc
- code is easier

### CODE REVIEW

File: Sources/Stats.php [hunk 1]
```diff
@@ -437,22 +437,21 @@ function DisplayStats()
...
 			'member_list' => array_keys($members),
-			'top_topic_posters' => implode(',', array_keys($members)),
 		)
 	);
 	$context['stats_blocks']['starters'] = array();
 	$max_num = 1;
 	while ($row_members = $smcFunc['db_fetch_assoc']($members_result))
 	{
-		$context['stats_blocks']['starters'][] = array(
+		$i = array_search($row_members['id_member'], array_keys($members));
```
Comment (live627) [inline Sources/Stats.php]: `$i = $members[$row_members['id_member']]`

File: Sources/Stats.php [hunk 2]
```diff
@@ -437,22 +437,21 @@ function DisplayStats()
...
+		$context['stats_blocks']['starters'][$i] = array(
 			'name' => $row_members['real_name'],
 			'id' => $row_members['id_member'],
 			'num' => $members[$row_members['id_member']],
```
Comment (live627) [inline Sources/Stats.php]: `'num' => $i`

File: Sources/Stats.php [hunk 3]
```diff
@@ -463,6 +462,7 @@ function DisplayStats()
 		if ($max_num < $members[$row_members['id_member']])
 			$max_num = $members[$row_members['id_member']];
 	}
+	ksort($context['stats_blocks']['starters']);
```
Comment (live627) [inline Sources/Stats.php]: This is backwards.

http://php.net/manual/en/function.krsort.php

**Comment by albertlast** [comment]
The first idea was,
to evade the whole sorting stuff and set the array on the right index,
but php didn't allow stuff like this.

So I take your advice.
I keep on line 453 with $i = ... , because otherwise the line 454 get to big.

**Comment by live627** [comment]
I think  I  foresee an issue with this. Two or more members with the same stat won't show right.

**Comment by live627** [comment]
Only one will show; the rest are clipped.

**Comment by albertlast** [comment]
I didn't noticed this too,
move back to the first solution.

**Comment by live627** [comment]
I haven't tested it, but it seems the first solution won't sort correctly either.

The solution here is to sort the array according to the order of $members.

**Comment by albertlast** [comment]
In my test area this works: http://snag.gy/BPxG3.jpg

**Pull request merged**